dali/pipeline/operators/decoder/nvjpeg/legacy_api/nvjpeg_decoder.h

```diff
@@ -236,6 +237,7 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
           const auto shape = image->GetShape();
           info.heights[0] = shape[0];
           info.widths[0] = shape[1];
+          nchannels = shape[2] > 0 ? shape[2] : nchannels;
```

Comment:
1. What happens if `shape[2] > 0 && shape[2] != 3` and output_type_ is DALI_RGB?
2. Can it be that the shape has 0 channels?
3. Also, if the image is `>3` channel but `output_type_` is still `DALI_RGB` - shouldn't we just take first 3 channels and hope for the best?

Reply:
1 and 3: You are right. I have to take care of that case. 
2: Yes, unfortunately some (most) of the implementations of `PeekShape` disregard the number of channels and return a `0`

I'll push a patch in a moment